### PR TITLE
hack: openstack: initial commit

### PR DESCRIPTION
**Pull request by philips**
This adds OpenStack instructions tested against vexxhost. Overall
everything seems to work except the kubernetes cloud provider integration for LBaaS but I think
this is a vexxhost issue.

### CODE REVIEW

**Comment by philips** [comment]
@aaronlevy can I just merge this? It has been tested so I would rather just have it than not.

File: hack/quickstart/openstack/config.tf [hunk 1]
```diff
@@ -0,0 +1,24 @@
+variable "key_pair" {
+  type    = "string"
+  default = "philips"
```
Comment (aaronlevy) [inline hack/quickstart/openstack/config.tf]: Use a more generic placeholder?

File: hack/quickstart/openstack/config.tf [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+variable "flavor_id" {
+  type    = "string"
+  default = "bbcb7eb5-5c8d-498f-9d7e-307c575d3566"
```
Comment (aaronlevy) [inline hack/quickstart/openstack/config.tf]: What are these values mapping to? Is this specific to a particular provider or openstack in general?

File: hack/quickstart/openstack/openstack.tf [hunk 3]
```diff
@@ -0,0 +1,102 @@
+resource "openstack_compute_secgroup_v2" "kubernetes_control_public" {
+  name = "80AC0F3A-ECF8-424C-88FA-D8571F0C0EC9"
```
Comment (aaronlevy) [inline hack/quickstart/openstack/openstack.tf]: Is this significant or just placeholder?

File: hack/quickstart/quickstart-openstack.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+Similarly on the controller manager:
+
+```
+kubectl edit deployment -n kube-system kube-controller-manager
+```
+
+```
+      volumes:
+      - hostPath:
+          path: /etc/kubernetes
```
Comment (aaronlevy) [inline hack/quickstart/quickstart-openstack.md]: We shouldn't be relying on host paths - this should ideally be in a configMap

**Comment by aaronlevy** [comment]
A couple minor comments.

One of my concerns is that I likely will not be able to adequately keep this updated as bootkube changes fairly rapidly right now (and I personally have no openstack experience).

Couple options:
- somehow mark it as "best effort maintained"
- owners files for each implementation
- move to a "contrib" directory (and note that it is "best effort" maintained)

**Comment by philips** [comment]
@aaronlevy just assign issues to me.

**Comment by philips** [comment]
@aaronlevy I will fix up the nits.

File: hack/quickstart/openstack/config.tf (hunk 1 above)
Comment (s-urbaniak) [inline hack/quickstart/openstack/config.tf]: This is just the default value which can be override in terraform using `terraform -var 'key_pair=mykey' apply/plan`, I'll update the readme accordingly.

File: hack/quickstart/openstack/config.tf (hunk 2 above)
Comment (s-urbaniak) [inline hack/quickstart/openstack/config.tf]: This seems to be provider specific, openstack comes with defaults, but vexxhost has its own. I'll add a small chapter on how to list flavors.

File: hack/quickstart/openstack/openstack.tf (hunk 3 above)
Comment (s-urbaniak) [inline hack/quickstart/openstack/openstack.tf]: I believe this is arbitrary and optionally used for back-references, did you auto-generate the names here @philips?

Comment (s-urbaniak) [inline hack/quickstart/openstack/openstack.tf]: The name is not even used for back-references, I'll humanize them.

File: hack/quickstart/quickstart-openstack.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+$ terraform apply openstack/
+```
+
+
+### Bootstrap Master
+
+*Replace* `<node-ip>` with the EXTERNAL_IP from output of `openstack server list --format=json --name 'control_node_.' | jq .[].Networks`.
+
+```
+export IDENT=~/.ssh/id_rsa.pub; ./init-master.sh <node-ip>
```
Comment (s-urbaniak) [inline hack/quickstart/quickstart-openstack.md]: Note to myself: replace and document this with the generated openstack keypair.

File: hack/quickstart/quickstart-openstack.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+```
+
+### Add Workers
+
+*Replace* `<node-ip>` with the EXTERNAL_IP from output of `openstack server list --format=json --name 'worker_node_.' | jq .[].Networks`.
+
+Initialize each worker node by replacing `<node-ip>` with the EXTERNAL_IP from the commands above.
+
+```
+$ IDENT=~/.ssh/google_compute_engine ./init-worker.sh <node-ip> cluster/auth/kubeconfig
```
Comment (s-urbaniak) [inline hack/quickstart/quickstart-openstack.md]: Note to myself: replace and document this with the generated openstack keypair.

**Comment by s-urbaniak** [comment]
I suggest to close this and proceed the discussion in #299

**Pull request closed without merging**